drivers/hwinfo/hwinfo_nrf.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2018 Alexander Wachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <soc.h>
+#include <hwinfo.h>
+#include <string.h>
+
+struct nrf_uid {
+	u32_t id[2];
+};
+
+static int nrf_id_init(struct device *dev)
```

Comment:
This is inconsistent with the API. I suggest you use `_init_id` and `_get_id` everywhere